### PR TITLE
migrated to use brew prefix.

### LINKED ISSUE #924: HOMEBREW_PREFIX not being honoured.
there are 9 cases in valet where "/usr/local/" is hardcoded in, rather than honouring HOMEBREW_PREFIX

https://github.com/laravel/valet/search?q=%2Fusr%2Flocal&unscoped_q=%2Fusr%2Flocal

This means that any installation of homebrew outside of /usr/local (not so unusual when dealing with multiple toolchains) requires manual edits to the valet codebase.

This is such an easy thing to fix, I don't really know why it hasn't been done.

### PR DESCRIPTION
**Pull request by MrBenGriffin**
I am sorry - I have not been able to test this properly.
However, it seems to be ok in general.

I failed to modify the Brew tests.

Closes #924

### CODE REVIEW

File: cli/Valet/Brew.php [hunk 1]
```diff
@@ -36,6 +36,26 @@ function __construct(CommandLine $cli, Filesystem $files)
     {
         $this->cli = $cli;
         $this->files = $files;
+        $this->homebrew_prefix = '/usr/local';
+        $this->prefix_loaded = false;
+    }
+
+    /**
+     * Determine HOMEBREW_PREFIX
```
Comment (drbyte) [inline cli/Valet/Brew.php]: In your modified environment, does calling `brew --prefix` work properly as a replacement for this custom function?

**Comment by drbyte** [comment]
@MrBenGriffin Thanks for this!

I hope to do some testing on it soon, and take a stab at what needs doing with the Brew tests.

When you have a minute, please edit your initial post above to add `Closes #924` so that the issue automatically closes when this gets merged. It just helps with housekeeping ;)

File: cli/Valet/Brew.php (hunk 1 above)
Comment (MrBenGriffin) [reply cli/Valet/Brew.php]: OMG. yep.  just use that!! "brew --prefix"

Comment (MrBenGriffin) [reply cli/Valet/Brew.php]: I modified the code to use "brew --prefix" internally. It was a better choice.
Internally brew uses cwd, so getting the path from which brew was more like what it does - but as it has a switch, we should use it.

**Comment by MrBenGriffin** [comment]
Bit confused - not hurrying - but just checking in - am I done on this?  If there's more, let me know.

**Comment by drbyte** [comment]
@MrBenGriffin I think so. I thought I'd have time yesterday to dig, but a work project came up that needed my full attention.

I might get a bit of time to explore it this afternoon.

**Comment by drbyte** [comment]
Observations:

- The Nginx.php `lint()` method needs `$prefix` inserted, else `valet secure` and `valet restart` won't work.

- Several tests will need updating: mostly to mock out the prefix so it can be tested against

**Comment by mattstauffer** [comment]
I believe this has been replaced with a recently merged PR, so closing. Thanks!

**Pull request closed without merging**